### PR TITLE
Indicate suicide reason with exit code. (#6469)

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
The current exit code `137` is confusing since it can indicate an
external `SIGKILL` or a Marathon suicide. This change introduces
different exit codes to indicate why Marathon stopped itself.

JIRA issues: MARATHON-8381

### CODE REVIEW

**Comment by timcharper** [comment]
I missed the original PR; can we please follow up with an update to changelog? This is noteworthy :)

**Comment by jeschkies** [comment]
> I missed the original PR; can we please follow up with an update to changelog? This is noteworthy :)

The original PR is https://github.com/mesosphere/marathon/pull/6469. I'll follow up with a changelog note. Thanks for the reminder.

File: docs/docs/exit-codes.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+| Exit Code | Reason                                                               |
+|-----------|----------------------------------------------------------------------|
+|100        | `ZookeeperConnectionFailure` - Could not connect to Zookeeper        |
+|101        | `ZookeeperConnectionLost` - Lost connect to Zookeeper                |
+|102        | `PluginInitializationFailure` - Could not load plugin                |
+|103        | `LeadershipLoss` - Lost leadership                                   |
+|104        | `LeadershipEndedFaulty` - Leadership ended with an error             |
+|105        | `LeadershipEndedGracefully` - Leadership ended without an error      |
+|106        | `MesosSchedulerError` - The Mesos scheduler driver had an error       |
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply docs/docs/exit-codes.md]: I think the driver throws an error. @zen-dog, knows better than I do.

[the remark this answers is not in the document]
Comment (zen-dog) [reply docs/docs/exit-codes.md]: Well, it's complicated, so here is a gist: driver doesn't fail, our own `HeartBeatMonitor` notices and eventually calls `driver.stop` which lands [here](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala#L253) which then calls `electionService.abdicateLeadership()` which is... wrong exit reason ^.^

[the remark this answers is not in the document]
Comment (zen-dog) [reply docs/docs/exit-codes.md]: But, yes, good catch @alenkacz , I think you should follow up with this change @jeschkies

**Pull request merged**